src/users/users.controller.spec.ts

```
@@ -0,0 +1,45 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { UsersController } from './users.controller';
+import { Role } from './types/role';
+import { UsersService } from './users.service';
+import { User } from './types/user.entity';
+
+const mockUser: User = {
+  id: 1,
+  email: 'test@test.com',
+  role: Role.ADMIN,
+  isClaimed: true,
+};
+
+const serviceMock: Partial<UsersService> = {
+  create: jest.fn(() => Promise.resolve(mockUser)),
+};
+
+describe('UsersController', () => {
+  let controller: UsersController;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      controllers: [UsersController],
+      providers: [
+        {
+          provide: UsersService,
+          useValue: serviceMock,
+        },
+      ],
+    }).compile();
+
+    controller = module.get<UsersController>(UsersController);
+  });
+
+  it('should delegate to the users service', async () => {
+    expect.assertions(2);
```

I forgot to mention this but, we want to use `expect.assertions(2)` to assert the number of expects called in async tests.

This is important because with async its super easy to forget to await and turn the result into a pending promise. When this happens the test will silently succeed but actually test nothing!